clear/main.py

```diff
@@ -79,20 +79,22 @@ def get_command(self, ctx, cmd_name):
 # location (configdb?), so that we prevent the continous execution of this
 # bash oneliner. To be revisited once routing-stack info is tracked somewhere.
 def get_routing_stack():
-    cmd0 = ["sudo", "docker", "ps"]
-    cmd1 = ["grep", "bgp"]
-    cmd2 = ["awk", '{print$2}']
-    cmd3 = ["cut", "-d-", "-f3"]
-    cmd4 = ["cut", "-d:", "-f1"]
+    result = 'frr'
+
+    cmd0 = ["sudo", "docker", "ps", "--format", "{{.Image}}\t{{.Names}}"]
+    cmd1 = ["awk", '$2 == "bgp"']
+    cmd2 = ["cut", "-d-", "-f3"]
+    cmd3 = ["cut", "-d:", "-f1"]
+    cmd4 = ["head", "-n", "1"]
 
     try:
         _, result = getstatusoutput_noshell_pipe(cmd0, cmd1, cmd2, cmd3, cmd4)
 
     except OSError as e:
-        raise OSError("Cannot detect routing-stack")
-
-    return (result)
+        # If detection fails, return "frr" as default
```

Comment:
>default [](http://example.com/codeflow?start=46&length=7)

This default value will hide bugs in above code. Suggest not to provide default value.

Reply:
Thanks for your reminder, @qiluo-msft , Actually, I tried to remove the check and return the frr directly at first, but caused failures in UT cases due to "add_command", So I updated the query cmd to match the exact bgp docker name, and set the default value to frr to avoid any potential issues. 
Please let me know if any other concerns, and please let me know if you want to keep the raising exception to catch the issue, thanks a lot.

Reply:
revert the commit and raise the exception as previous to catch the issue in advance, thanks

Reply:
you raise another new OSError object, and lose original `e`. Could you just `raise` without object?

Reply:
"I do not understand why "but caused failures in UT cases due to "add_command"""
---It's for the first solution, I tried to remove the quagga related code, but failed with UT case, seems related to the dynamic python module import, quagga or frr.